docs/storage-backend/cassandra.md

```diff
@@ -194,37 +194,41 @@ iterate over all vertices or edges in large graphs effectively.
 >
 > —  [DataStax Astra](https://www.datastax.com/products/datastax-astra)
 
-[Download](https://docs.datastax.com/en/astra/aws/doc/dscloud/astra/dscloudObtainingCredentials.html) the secure-connect zipped bundle for your Astra database.
-
-Unzip your copy of `secure-connect-your_astra_db.zip` which will contain the following files:
-
-- `ca.crt`
-- `cert`
-- `cert.pfx`
-- `config.json`
-- `cqlshrc`
-- `identity.jks`
-- `key`
-- `trustStore.jks`
-
-The `config.json` file contains the following:
-
-- `host` - the contact point for the CQL driver
-- `port` - the CQL client port
-- `keyspace` - the name of the keyspace you created
-- `trustStoreLocation` - trustStore.jks is the truststore
-- `trustStorePassword` - your truststore password
-- `keyStoreLocation` - identity.jks is the keystore
-- `keyStorePassword` - your keystore password
-
-Using the information above, you will need to configure the following:
-
-- `storage.hostname`
-- `storage.port`
-- `storage.username`
-- `storage.password`
-- `storage.cql.keyspace`
-- `storage.cql.ssl.*`
+[Download](https://docs.datastax.com/en/astra/docs/obtaining-database-credentials.html) the secure-connect zipped bundle for your Astra database.
+
+While connecting to Astra DB from JanusGraph, it is preferred to make use of the setting 
+`storage.cql.internal.file-configuration` and externalize the astra connection related properties to a file,
+since the datastax driver used internally knows how to work with the secure connection bundle and connect
+to the Astra DB
+
+Here's a minimal setting required to connect to Astra DB:
+
+janus.properties
+```properties
+gremlin.graph=org.janusgraph.core.JanusGraphFactory
+storage.backend=cql
+storage.cql.keyspace=<keyspace-name>
+
+# Link to the external file that DataStax driver understands
+storage.cql.internal.file-configuration=<path-to-astra.conf>
+```
+
+astra.conf
+```
+datastax-java-driver {
+  basic.cloud {
+    secure-connect-bundle = "<path-to-secure-bundle-zip-file"
+  }
+  advanced.auth-provider {
+    class = PlainTextAuthProvider
+    username = "<clientID>"
+    password = "<clientSecret>"
+  }
+}
+```
+
+_Client id and Client secret need to be generated and copied from your Astra Account
+as per this [doc](https://docs.datastax.com/en/astra/docs/manage-application-tokens.html)_
 
```

Comment:
Not necessary but I would also add here something like the below:

----
If you want to keep DataStax driver configuration inside the JanusGraph configuration you can make use of `storage.cql.internal.string-configuration` instead of `storage.cql.internal.file-configuration`:
 ```
storage.backend: cql
storage.hostname: <astradb hostname which can be found inside secure-connect-your_astra_db.zip in config.json file>
storage.port: <astradb port which can be found inside secure-connect-your_astra_db.zip in config.json file>
storage.cql.keyspace: <keyspace name which was created in AstraDB>
storage.username: <clientID>
storage.password: <clientSecret>
storage.cql.local-datacenter: <astradb local datacenter which can be found inside secure-connect-your_astra_db.zip in config.json file>
storage.cql.internal.string-configuration=datastax-java-driver { basic.cloud.secure-connect-bundle=<path-to-secure-bundle-zip-file> }
```

Reply:
Makes sense @porunov. I have trimmed the config even more and tested. it looks like below (Didn't have to unzip the secure bundle file at all)
```
gremlin.graph=org.janusgraph.core.JanusGraphFactory
storage.backend=cql
storage.cql.keyspace=<keyspace name which was created in AstraDB>
storage.username=<clientID>
storage.password=<clientSecret>
storage.cql.internal.string-configuration=datastax-java-driver { basic.cloud.secure-connect-bundle=<path-to-secure-bundle-zip-file> }
```
Also, another option to provide the secure bundle file instead of `storage.cql.internal.string-configuration` property is that we can pass vm arg as follows: `-Ddatastax-java-driver.basic.cloud.secure-connect-bundle=<path-to-secure-bundle-zip-file>`

Hope the above looks good, let me know your comments.

I'll update the doc with above details if all look good, thanks

Reply:
It looks good to me. Thank you @kathirsvn !
Just make sure to squash your commits into a single commit and sign it with `-s` as you did it with your first commit.